src/chapters/typescript/functions.rst

```diff
@@ -0,0 +1,115 @@
+Functions in TypeScript
+=======================
+
+When creating functions in TypeScript, we have many of the same options as in JavaScript.
+We can make anonymous functions, give it a different number of parameters and so on.
+However, when working with functions in TypeScript, we have to keep *type* in mind.
+
+Declaring Functions
+-------------------
+
+Named Functions
+^^^^^^^^^^^^^^^
+
+Let's take a look at a TypeScript function declaration.
+
+.. sourcecode:: typescript
+
+   function myFunction(a: number): number {
+      return a*2;
+   }
+
+Here you can see that we provided the type of the parameters and the type of the value that is returned.
+If the function doesn't have a return value, then you use ``void`` for its type.
+
+.. note::
+
+   While in many cases, it is possible to determine the return type during compilation, if you know the return type, you should give it. 
+
+Anonymous Functions
+^^^^^^^^^^^^^^^^^^^
+
+For an anonymous function, you still need to provide types for the value returned and the parameters.
+
+.. sourcecode:: typescript
+
+   let myFunction = function(a: number): number { return a*2; };
+
+The above code looks very much like an anonymous function declaration with the types included!
+However, TypeScript has arrow notation and here we use the **fat arrow**, ``=>``.
+
+.. sourcecode:: typescript
+
+   let myFunction: (a: number) => number = function(a: number): number { return a*2; };
+
+In TypeScript, the keywords that come after the colon are the type. The ``=>`` operator then indicates the return type of the function.
+
+Optional Parameter
+------------------
+
+In JavaScript, you can declare a function with 5 parameters and only give it 2 when it comes time to use it.
+In TypeScript, that is not the case. When you declare a function with 5 parameters, you have to give the function 5 arguments when calling it.
+However, in TypeScript, you can make parameters optional. This means that when you are calling the function, you can leave off the optional parameter.
```

Comment:
i think combining this sentence and the one before it makes this concept a bit clearer for me. 

"In TypeScript when you declare a function with 5 parameters, you have to give the function 5 arguments when calling it, unless you make some of those parameters optional. As with JavaScript functions, optional parameters can be omitted."

Reply:
Done! Thanks!